packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -68,7 +69,10 @@ export default function createMiddleware(config: MiddlewareConfig) {
     }
 
     function rewrite(url: string) {
-      return NextResponse.rewrite(new URL(url, request.url), getResponseInit());
```

Comment:
I think the problem here is the construction with `request.url`. If we apply `request.pathname` then maybe we can avoid introducing more conditions.